### PR TITLE
Fix on station jet air using and trails only on use.

### LINKED ISSUE #48661: Inertia flying consume jetpack air.
BYOND Version: 513.1505
Server revision compiled on: 2020-01-07
Master commit: e87033d792cf4834f4fc18b0e6a2c2057859ce89

## Reproduction:

Equip jetpack turn on (with no stabilization) and jump to graveless room. 
You see as you fly doing nothing you jetpack use air.

### PR DESCRIPTION
**Pull request by Dennok**
## About The Pull Request

Jet backpack and implant use air only when it must be used.

## Why It's Good For The Game

No more persistent jet air use.
fix #48661

close #49569

No trails on inertia space move. No gasps in trail on jet flying.

trails

I tested it and seems it work nice.

## Changelog
:cl:
tweak: Now spacemans don't try push things around to propel yourself, when use jetpack.
fix: Now jetpacks don't use air in gravity area or when you just inertia flying in space.
tweak: Now jetpacks generate trails only on air use.
/:cl:

### CODE REVIEW

**Comment by TheVekter** [comment]
I don't wanna be that guy, but can you work out the grammar issues in this PR? I can guess based on the issue it closes what it's meant to do, but the changelog makes zero sense.

**Comment by Dennok** [comment]
> I don't wanna be that guy, but can you work out the grammar issues in this PR? I can guess based on the issue it closes what it's meant to do, but the changelog makes zero sense.

Better?

**Comment by TheVekter** [comment]
Yeah, better.

**Comment by cacogen** [comment]
good fix

File: code/game/objects/items/tanks/jetpack.dm [hunk 1]
```diff
@@ -67,7 +69,20 @@
...
 
 /obj/item/tank/jetpack/proc/move_react(mob/user)
-	allow_thrust(0.01, user)
+	if(!on)//If jet dont work, it dont work
+		return
+	if(!user)//Don't allow jet self using
+		return
+	if(!isturf(user.loc))//You can't use jet in nowhere or from mecha/closet
+		return
+	if(has_gravity(user))//You don't want use jet in gravity
```
Comment (ShizCalev) [inline code/game/objects/items/tanks/jetpack.dm]: !(L.is_flying() || L.is_floating()) || L.buckled

Comment (ShizCalev) [inline code/game/objects/items/tanks/jetpack.dm]: `!(L.is_flying() || L.is_floating()) || L.buckled`

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/modules/surgery/organs/augments_chest.dm [hunk 2]
```diff
@@ -171,7 +171,20 @@
...
 
 /obj/item/organ/cyberimp/chest/thrusters/proc/move_react()
-	allow_thrust(0.01)
+	if(!on)//If jet dont work, it dont work
+		return
+	if(!owner)//Don't allow jet self using
+		return
+	if(!isturf(owner.loc))//You can't use jet in nowhere or in mecha/closet
+		return
+	if(has_gravity(owner))//You don't want use jet in gravity
```
Comment (ShizCalev) [inline code/modules/surgery/organs/augments_chest.dm]: same here.

File: code/modules/surgery/organs/augments_chest.dm [hunk 3]
```diff
@@ -171,7 +171,20 @@
 		icon_state = "imp_jetpack"
```
Comment (ShizCalev) [inline code/modules/surgery/organs/augments_chest.dm]: toggle doesn't turn off the jetpack if allow_thrust is false like you did with the other one

File: code/game/objects/items/tanks/jetpack.dm (hunk 1 above)
Comment (Dennok) [reply code/game/objects/items/tanks/jetpack.dm]: You know that is_floating() in space can be negated by magboots?

Comment (Dennok) [reply code/game/objects/items/tanks/jetpack.dm]:
```suggestion
	if(!(user.is_flying() || user.is_floating()) || user.buckled)//You don't want use jet in gravity or while buckled.
```

File: code/modules/surgery/organs/augments_chest.dm (hunk 2 above)
Comment (Dennok) [reply code/modules/surgery/organs/augments_chest.dm]:
```suggestion
	if(!(owner.is_flying() || owner.is_floating()) || owner.buckled)//You don't want use jet in gravity or while buckled.
```

File: code/modules/surgery/organs/augments_chest.dm (hunk 3 above)
Comment (Dennok) [reply code/modules/surgery/organs/augments_chest.dm]: Ups, forgot about this one.

Comment (Dennok) [reply code/modules/surgery/organs/augments_chest.dm]: But this be toggled off after one try to move anyway.

File: code/modules/surgery/organs/augments_chest.dm [hunk 4]
```diff
@@ -148,8 +148,8 @@
 			if(!silent)
 				to_chat(owner, "<span class='warning'>Your thrusters set seems to be broken!</span>")
 			return 0
-		on = TRUE
 		if(allow_thrust(0.01))
```
Comment (ShizCalev) [inline code/modules/surgery/organs/augments_chest.dm]: early return

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/modules/surgery/organs/augments_chest.dm (hunk 4 above)
Comment (Dennok) [reply code/modules/surgery/organs/augments_chest.dm]: move check or what?

Comment (Dennok) [reply code/modules/surgery/organs/augments_chest.dm]: Fixed

**Pull request merged**